module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp

```diff
@@ -122,118 +130,191 @@ namespace module::behaviour::planning {
+
+                    case message::behaviour::MotionCommand::Type::DIRECT_COMMAND: walk_directly(); return;
+
+                    case message::behaviour::MotionCommand::Type::BALL_APPROACH: vision_walk_path(); return;
+
+                    case message::behaviour::MotionCommand::Type::WALK_TO_STATE: vision_walk_path(); return;
 
-                Eigen::Affine3d Htw(sensors.Htw);
-
-                auto now = NUClear::clock::now();
-                float timeSinceBallSeen =
-                    std::chrono::duration_cast<std::chrono::nanoseconds>(now - timeBallLastSeen).count()
-                    * (1.0f / std::nano::den);
-
-
-                Eigen::Vector3d rBWw_temp(ball.position.x(), ball.position.y(), fieldDescription.ball_radius);
-                rBWw     = timeSinceBallSeen < search_timeout ? rBWw_temp :                         // Place last seen
-                           Htw.inverse().linear().leftCols<1>() + Htw.inverse().translation();  // In front of the robot
-                position = (Htw * rBWw).head<2>();
-
-                // Hack Planner:
-                float headingChange = 0;
-                float sideStep      = 0;
-                float speedFactor   = 1;
-                if (useLocalisation) {
-
-                    // Transform kick target to torso space
-                    auto fieldPosition = Eigen::Affine2d(field.position);
-                    Eigen::Affine3d Hfw;
-                    Hfw.translation() =
-                        Eigen::Vector3d(fieldPosition.translation().x(), fieldPosition.translation().y(), 0);
-                    Hfw.linear() = Eigen::AngleAxisd(Eigen::Rotation2Dd(fieldPosition.rotation()).angle(),
-                                                     Eigen::Vector3d::UnitZ())
-                                       .toRotationMatrix();
-
-                    Eigen::Affine3d Htf        = Htw * Hfw.inverse();
-                    Eigen::Vector3d kickTarget = Htf * Eigen::Vector3d(kickPlan.target.x(), kickPlan.target.y(), 0);
-
-                    // //approach point:
-                    Eigen::Vector2d ballToTarget = (kickTarget.head<2>() - position).normalized();
-                    Eigen::Vector2d kick_point   = position - ballToTarget * ball_approach_dist;
-
-                    if (position.norm() > slowdown_distance) {
-                        position = kick_point;
-                    }
-                    else {
-                        speedFactor   = slow_approach_factor;
-                        headingChange = std::atan2(ballToTarget.y(), ballToTarget.x());
-                        sideStep      = 1;
-                    }
+                    case message::behaviour::MotionCommand::Type::ROTATE_ON_SPOT: rotate_on_spot(); return;
+
+                    case message::behaviour::MotionCommand::Type::WALK_TO_READY: walk_to_ready(); return;
+                    // This line should be UNREACHABLE
+                    default:
+                        log<NUClear::WARN>(
```

Comment:
I would make this an error, it really shouldn't happen.

Reply:
Should I throw error like this:
`throw std::runtime_error(
                            fmt::format("Invalid walk path planning command {}.", latestCommand.type.value));`

Reply:
I'd just change it from `NUClear::WARN` to `NUClear::ERROR`

Reply:
Ah I see.

Reply:
```suggestion
                        log<NUClear::ERROR>(
```